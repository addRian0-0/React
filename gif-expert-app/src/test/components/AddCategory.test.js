import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('pruebas en addCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('simular cambios en el input', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {
            target: {
                value
            }
        });

    })

    test('NO debe de postear la informaciÃ³n con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();

    })

    test('debe de llamar el SetCategories y limpiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', {
            target: {
                value
            }
        });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalled();
        expect( wrapper.find('input').prop('value') ).toBe('');

    })


})