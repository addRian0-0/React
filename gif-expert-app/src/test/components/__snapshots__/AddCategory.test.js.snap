// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en addCategory debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
